bluesky/run_engine.py

```diff
@@ -1377,18 +1379,21 @@ def _read(self, msg):
         if self._bundling:
             # if the object is not in the _describe_cache, cache it
             if obj not in self._describe_cache:
+                self.describe_cache[obj] = dict()
```

Comment:
There is a nice trick we use in a couple places would make these two lines unneeded (Lines 1381 and 1382, as of this writing). In `RunEngine.__init__`, make `self._describe_cache` a `defaultdict(dict)` instead of a plain `dict`. Then you can check `if 'regular' not in self._describe_cache[obj]:` directly, knowing that `self._describe_cache[obj]` will be magically initialized to `dict()` if it needs to be.